discovery-infra/tests/base_test.py

```diff
@@ -34,29 +38,105 @@
 
 
 class BaseTest:
+
+    @pytest.fixture
+    def controller_configuration(self):
+        if global_variables.platform == consts.Platforms.VSPHERE:
+            return VSphereControllerConfig()
+
+        return TerraformConfig()
+
+    @pytest.fixture
+    def prepare_controller_configuration(self, controller_configuration):
+        if not isinstance(controller_configuration, TerraformConfig):
+            return controller_configuration
+
+        net_asset = LibvirtNetworkAssets()
+        tf_config = TerraformConfig()
+        tf_config.net_asset = net_asset.get()
+        yield tf_config
+        net_asset.release_all()
+
+    @pytest.fixture
+    def cluster_configuration(self):
+        return ClusterConfig()
+
+    @pytest.fixture
+    def controller(self, cluster_configuration, prepare_controller_configuration):
+        if cluster_configuration.platform == consts.Platforms.VSPHERE:
+            return VSphereController(prepare_controller_configuration, cluster_configuration)
+
+        return TerraformController(prepare_controller_configuration, cluster_config=cluster_configuration)
+
+    @pytest.fixture
+    def nodes(self, controller):
+        return Nodes(controller)
+
+    @pytest.fixture
+    def prepare_nodes(self, nodes, cluster_configuration):
+        nodes.prepare_nodes()
+        yield nodes
+        if global_variables.test_teardown:
+            logging.info('--- TEARDOWN --- node controller\n')
+            nodes.destroy_all_nodes()
+            logging.info(f'--- TEARDOWN --- deleting iso file from: {cluster_configuration.iso_download_path}\n')
+            infra_utils.run_command(f"rm -f {cluster_configuration.iso_download_path}", shell=True)
+
+    @pytest.fixture
+    def prepare_network(self, prepare_nodes, prepare_controller_configuration):
+        if global_variables.platform != consts.Platforms.BARE_METAL and global_variables.platform != consts.Platforms.NONE:
+            return prepare_nodes
+
+        interfaces = BaseTest.nat_interfaces(prepare_controller_configuration)
+        nat = NatController(interfaces, NatController.get_namespace_index(interfaces[0]))
+        nat.add_nat_rules()
+        yield prepare_nodes
+        if nat:
+            nat.remove_nat_rules()
+
+    @pytest.fixture
+    def cluster(self, api_client, request, proxy_server, prepare_network, cluster_configuration):
```

Comment:
While `prepare_network` is the last fixture in the chain it says nothing about what the fixture does.

Reply:
prepare network is preparing the network.... in this case configuring the NAT. don't understand what do you mean by "says nothing about what the fixture does" can you please extend?